activiti-cloud-services-runtime-bundle/activiti-cloud-services-subscriptions/src/main/java/org/activiti/services/subscriptions/behavior/BroadcastSignalEventActivityBehavior.java

```diff
@@ -0,0 +1,43 @@
+package org.activiti.services.subscriptions.behavior;
+
+import org.activiti.bpmn.model.Signal;
+import org.activiti.bpmn.model.SignalEventDefinition;
+import org.activiti.cloud.services.api.commands.SignalCmd;
+import org.activiti.engine.delegate.DelegateExecution;
+import org.activiti.engine.delegate.Expression;
+import org.activiti.engine.impl.bpmn.behavior.IntermediateThrowSignalEventActivityBehavior;
+import org.activiti.engine.impl.context.Context;
+import org.activiti.engine.impl.interceptor.CommandContext;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+@Scope("prototype")
```

Comment:
@daisuke-yoshimoto why do we need this?

Reply:
@Salaboy 
This class is not a singleton, so we need to annotate this class by @Scope("prototype").
Every time a process definition is deployed, another instance of this class is created.
Each instance holds the setting information of this signal as a member variable.

Reply:
@daisuke-yoshimoto I'm trying to understand this.. why do we need different instances? Is that the default behaviour for the engine with The SignalEventActivitiBehaviour? Does it have multiple instances? If not.. we should just have a singleton instance that deals with sending all messages for all signals.

Reply:
@Salaboy this activity behavior cannot be shared between process definitions since it holds the process definition's specific signal event definition. Yes, this is how the engine works

Reply:
@balsarori yeah.. now I've learnt that :)